### PR TITLE
feat: persist application logs

### PR DESCRIPTION
**Pull request by cristianoventura**
## Description

This PR introduces changes needed to capture caught and uncaught exceptions to a log file. This will enable us to troubleshoot issues more efficiently.

Here's an example of a log file being generated by the Studio. This includes errors raised by the proxy, IpcMain call and unhandle exceptions.

```log
[2024-10-07 12:06:39.823] [error] mitmproxy/certs.py:112: CryptographyDeprecationWarning: Properties that return a naïve datetime object have been deprecated. Please switch to not_valid_after_utc.

[2024-10-07 12:06:39.824] [error] /Users/username/www/k6-studio/resources/mac/arm64/k6-studio-proxy: Invalid proxy mode specification: regular@AAA (invalid port: AAA)

[2024-10-07 12:08:10.704] [error] Error: Error invoking remote method 'har:open': Error: ENOENT: no such file or directory, open '/Users/username/Documents/k6-studio/Recordings/2024-10-02_17-21-55.har'

[2024-10-07 12:17:58.722] [error] Unhandled rejection ReferenceError: currentBrowserProcess_NOT_DECLARED is not defined
    at IpcMainImpl.<anonymous> (/Users/username/www/k6-studio/.vite/build/main.js:2391:3)
    at IpcMainImpl.emit (node:events:518:28)
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:82429)
    at WebContents.emit (node:events:518:28)
```

In addition, a button was included in the UI to allow users to retrieve the log file more easily.

## How to Test

- Manually set the proxy to an invalid port and start the application
- Manually change the recording path of a generator and try to open it
- Throw exceptions in specific parts of the app and raise them from the UI

## Checklist

- [X] I have performed a self-review of my code.
- [X] I have added tests for my changes.
- [X] I have run linter locally (`npm run lint`) and all checks pass.
- [X] I have run tests locally (`npm test`) and all tests pass.
- [X] I have commented on my code, particularly in hard-to-understand areas.

## Screenshots (if appropriate):

image

## Related PR(s)/Issue(s)

Resolves https://github.com/grafana/k6-studio/issues/235

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,15 @@ To do that, modify the `hosts` file on your system, for example, `127.0.0.1 myap
...
 If you're on a Mac, make sure you're not running the k6 Studio application from the Downloads folder. If that's the case, close the app, move the application file to the Applications folder, and start the app again.
 
+### Application logs
+
+Application logs are saved in the following directory:
+
+- on Mac: `~/Library/Logs/k6 Studio/k6-studio.log`
+- on Windows: `%USERPROFILE%\AppData\Roaming\k6 Studio\logs\k6-studio.log`
+
+When opening an issue, please include a tail of your log file.
```
Comment (Llandy3d) [inline README.md:101]: We should probably add a section in the bug template asking for this 🤔

File: src/logger.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { spawn } from 'node:child_process'
+import path from 'node:path'
+
+export function initializeLogger() {
+  // allow logs to be triggered from the renderer process
+  // @see https://github.com/megahertz/electron-log/blob/master/docs/initialize.md
+  log.initialize()
+
+  // log electron core events
+  // @see https://github.com/megahertz/electron-log/blob/master/docs/events.md
```
Comment (Llandy3d) [inline src/logger.ts:11]: minor nit: I think the `@see` are redundant and probably the link alone would suffice 🤔

File: src/main.ts [hunk 3]
```diff
@@ -505,6 +511,11 @@ ipcMain.handle('browser:open:external:link', (_, url: string) => {
   shell.openExternal(url)
 })
 
+ipcMain.handle('application:open-log', () => {
```
Comment (Llandy3d) [inline src/main.ts:514]: minor suggestion: should we shorten it with `app:open-log` ?
I'm not sure about this

**Review by e-fisher — COMMENTED** [review]
> Throw exceptions in specific parts of the app and raise them from the UI

Should it catch UI exceptions automatically? I've tried adding `throw new Error('test')` to `Home.tsx` but it wasn't logged. 

If I misunderstood, and it wasn't supposed to catch UI exceptions, could we wrap App in error boundary to implement this?

**Comment by cristianoventura** [comment]
> Should it catch UI exceptions automatically? I've tried adding `throw new Error('test')` to `Home.tsx` but it wasn't logged.
> 
> If I misunderstood, and it wasn't supposed to catch UI exceptions, could we wrap App in error boundary to implement this?

That was missing since I forgot to initialize the error handler in the UI, good catch! Now it should be logging those to the disk as well! The error boundary is something we're missing and would be a nice addition but since we have everything logged to disk at this point, I'll leave the error boundary for a follow-up PR.

**Review by Llandy3d — APPROVED** [review]
🙌

**Review by e-fisher — APPROVED** [review]
LGTM! 🚀

**Pull request merged**